c10/util/order_preserving_flat_hash_map.h

```diff
@@ -699,22 +707,28 @@ class sherwood_v3_table : private EntryAlloc, private Hasher, private Equal
     {
         if (begin_it == end_it)
             return { begin_it.current };
-        for (EntryPointer it = begin_it.current, end = end_it.current; it != end; ++it)
-        {
-            if (it->has_value())
-            {
-                it->destroy_value();
-                --num_elements;
-            }
+
+        for (EntryPointer it = begin_it.current, end = end_it.current; it != end;) {
+          auto next = it->next;
+          remove_from_list(it);
+          it->destroy_value();
+          it = next;
```

Comment:
What happend to the num_elements adjustment here?

Reply:
good catch